### PR TITLE
WFLY-10083 Add runtime hot class replacement to WildFly

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Initial PR for testing+review, at this stage this is targeting WF14.

https://issues.jboss.org/browse/WFLY-10083

### CODE REVIEW

**Comment by bstansberry** [comment]
All the config stuff done for the legacy feature packs and dists will need to be done for galleon too.

**Comment by stuartwdouglas** [comment]
Retest this please

**Comment by stuartwdouglas** [comment]
Retest this please

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java [hunk 1]
```diff
@@ -442,6 +452,41 @@ private static void deployPersistenceUnit(
         }
     }
 
+    private static void handleClassChange(PersistenceUnitServiceImpl service, DeploymentClassChangeSupport classChangeSupport) {
+        if(classChangeSupport == null) {
+            return;
+        }
+        classChangeSupport.addListener(new ClassChangeListener() {
+            @Override
+            public void classesReplaced(List<ChangedClasssDefinition> replacedClasses, List<NewClassDefinition> newClassDefinitions) {
```
Comment (scottmarlow) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/PersistenceUnitServiceHandler.java]: what should the ordering sequence be for:
- Updated class definitions being loaded by classloader.
- classReplaced() being called.
- Weld rewriting application classes.
- Persistence provider [rewriting application (entity) classes](https://docs.oracle.com/javaee/7/api/javax/persistence/spi/ClassTransformer.html).

File: jpa/subsystem/src/main/java/org/jboss/as/jpa/service/EntityManagerFactoryDelegate.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import javax.persistence.EntityGraph;
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceUnitUtil;
+import javax.persistence.Query;
+import javax.persistence.SynchronizationType;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.metamodel.Metamodel;
+
+public class EntityManagerFactoryDelegate implements EntityManagerFactory {
```
Comment (scottmarlow) [inline jpa/subsystem/src/main/java/org/jboss/as/jpa/service/EntityManagerFactoryDelegate.java]: Where is the EntityManagerFactoryDelegate created?

**Comment by stuartwdouglas** [comment]
Retest this please

**Comment by kabir** [comment]
@stuartwdouglas I am closing this until it looks like this has some activity

**Pull request closed without merging**